src/java/org/streamingpool/core/service/impl/TrackKeepingDiscoveryService.java

```diff
@@ -118,7 +130,8 @@ private void checkSameContexOfExecution() {
 
             if (factoryResult.isPresent()) {
                 LOGGER.info(format("Stream from id '%s' was successfully created by factory '%s'", newId, factory));
-                return factoryResult;
+                Flowable<T> sharedDataStream = Flowable.fromPublisher(factoryResult.data());
```

Comment:
This flowable is unnecessary, or? Why not returning the factoryResult directly`?